### PR TITLE
[SER-448] Implement State_String in C#

### PR DESCRIPTION
**Pull request by tomthecarrot**

### CODE REVIEW

File: crates/rsharp/cs/src/RSharp.cs [hunk 1]
```diff
@@ -7,9 +7,9 @@ namespace RSharp
     public class Metadata
     {
 #if UNITY_IOS && !UNITY_EDITOR
-        internal const string LIBRARY_NAME = "__Internal";
+        public const string LIBRARY_NAME = "__Internal";
```
Comment (TheButlah) [inline crates/rsharp/cs/src/RSharp.cs]: Is public actually necessary?

File: crates/rsharp/cs/test/RBox.cs [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
+        {
+            this.output = output;
+        }
+
         [Fact]
         public void TestU8()
         {
+            output.WriteLine("Starting RBox U8 test");
             var u8 = new RBox_U8(10);
+            output.WriteLine("Finished RBox U8 test");
```
Comment (TheButlah) [inline crates/rsharp/cs/test/RBox.cs]:
```suggestion
```

**Review by TheButlah — CHANGES_REQUESTED** [review]

File: crates/rsharp/cs/src/RSharp.cs (hunk 1 above)
Comment (tomthecarrot) [reply crates/rsharp/cs/src/RSharp.cs]: No; changed back.

File: crates/rsharp/cs/test/RBox.cs (hunk 2 above)
Comment (tomthecarrot) [reply crates/rsharp/cs/test/RBox.cs]: Included this suggestion with another commit.

**Review by TheButlah — APPROVED** [review]

**Pull request merged**